### PR TITLE
V1.4: Full coverage test cases for `Storage` package

### PR DESCRIPTION
**Pull request by SeeuSim**
- [x] Refactor Storage I/O Managers into their own classes for brevity

## **Final Stats**

| Class Coverage | Method Coverage | Line Coverage |
|--|--|--|
| 100% (32/32) | 95% (154/162) | 88% (626/704) |

**~To add~ Added test cases for the following classes:**

- [x] jsonadapters (#256 )
  - [x] ingredient
  - [x] name
  - [x] duration
  - [x] portion
  - [x] portionunit
  - [x] step
  - [x] substitutionIngredient
  - [x] tag
  - [x] timeUnit
- [x] I/O Managers, if TestFX and CI permits
  - [x] ImportManager
  - [x] ExportManager

### CODE REVIEW

File: src/test/java/seedu/recipe/storage/filemanagers/ExportManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+     * Create the parent directory to host these test files.
+     */
+    @BeforeEach
+    public void createDir() {
+        try {
+            Files.createDirectory(TEMP_FOLDER);
+        } catch (IOException e) {
+            //do nothing
+            //TODO: Exit this test class
+        }
```
Comment (JamesLiuZX) [inline src/test/java/seedu/recipe/storage/filemanagers/ExportManagerTest.java:143]: assumeTrue

File: src/test/java/seedu/recipe/storage/jsonadapters/JsonAdaptedPortionUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import org.junit.jupiter.api.Test;
+
+import seedu.recipe.commons.exceptions.IllegalValueException;
+import seedu.recipe.model.recipe.unit.PortionUnit;
+
+//@@author alson001
+public class JsonAdaptedPortionUnitTest {
+    private static final String VALID_UNIT = "grams";
+    private static final String INVALID_UNIT = "";
+    private static final String NULL_UNIT = null;
```
Comment (JamesLiuZX) [inline src/test/java/seedu/recipe/storage/jsonadapters/JsonAdaptedPortionUnitTest.java:15]: null is tested in jsonadapted portion but not the rest, can standardise further

File: src/test/java/seedu/recipe/storage/jsonadapters/JsonAdaptedTagTest.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+import org.junit.jupiter.api.Test;
+
+import seedu.recipe.commons.exceptions.IllegalValueException;
+import seedu.recipe.model.tag.Tag;
+
+//@@author alson001
+public class JsonAdaptedTagTest {
+    private static final String VALID_TAG_NAME = "healthy";
+    private static final String INVALID_TAG_NAME = "#fatty";
```
Comment (JamesLiuZX) [inline src/test/java/seedu/recipe/storage/jsonadapters/JsonAdaptedTagTest.java:14]: Could test with maybe #f@tty/

File: src/test/java/seedu/recipe/model/recipe/RecipeDurationTest.java [hunk 4]
```diff
@@ -1,98 +1,66 @@
...
 
 import java.util.Objects;
 
 import org.junit.jupiter.api.Test;
 
-import seedu.recipe.model.recipe.exceptions.RecipeDurationInvalidArgumentLengthException;
-import seedu.recipe.model.recipe.exceptions.RecipeDurationInvalidDurationException;
 import seedu.recipe.model.recipe.unit.TimeUnit;
 
 public class RecipeDurationTest {
```
Comment (JamesLiuZX) [inline src/test/java/seedu/recipe/model/recipe/RecipeDurationTest.java:12]: Could add in multiple whitespace like "         " and null testing.

File: src/test/java/seedu/recipe/model/recipe/RecipePortionTest.java [hunk 5]
```diff
@@ -1,73 +1,59 @@
...
 import static seedu.recipe.testutil.Assert.assertThrows;
 
 import java.util.Objects;
 
 import org.junit.jupiter.api.Test;
 
-import seedu.recipe.model.recipe.exceptions.RecipePortionInvalidArgumentException;
 import seedu.recipe.model.recipe.unit.PortionUnit;
 
 public class RecipePortionTest {
```
Comment (JamesLiuZX) [inline src/test/java/seedu/recipe/model/recipe/RecipePortionTest.java:12]: Same as above, can add in multiple whitespace "       " and null

**Review by JamesLiuZX — COMMENTED** [review]
Great work! High effort test cases, present throughout model and storage.

File: src/test/java/seedu/recipe/model/recipe/RecipePortionTest.java (hunk 5 above)
Comment (SeeuSim) [reply src/test/java/seedu/recipe/model/recipe/RecipePortionTest.java:12]: for Model PR

File: src/test/java/seedu/recipe/model/recipe/RecipeDurationTest.java (hunk 4 above)
Comment (SeeuSim) [reply src/test/java/seedu/recipe/model/recipe/RecipeDurationTest.java:12]: for Model PR

File: src/test/java/seedu/recipe/storage/jsonadapters/JsonAdaptedPortionUnitTest.java (hunk 2 above)
Comment (SeeuSim) [reply src/test/java/seedu/recipe/storage/jsonadapters/JsonAdaptedPortionUnitTest.java:15]: standardised

File: src/test/java/seedu/recipe/storage/filemanagers/ExportManagerTest.java (hunk 1 above)
Comment (SeeuSim) [reply src/test/java/seedu/recipe/storage/filemanagers/ExportManagerTest.java]: fixed, test cases should proceed

**Pull request merged**